test/cpp/tensorexpr/tutorial.cpp

```diff
@@ -0,0 +1,426 @@
+    // expressions for each computation (which can be accessed via
+    // `func->func_var(idx)`) with the names specified by the first parameter of
+    // the constructor call. In our example two `Buf` variables will be created
+    // with names 'X' and 'Y', each of them would signify a domain of 64x32.
+
+    // We can now print out our function:
+    std::cout << "Tensor function: " << *func << std::endl;
+    // Prints:
+    // Tensor function: Function F(i[64], j[32]) {
+    //   X = i * j
+    //   Y = i + j
+    // }
+
+    // A Tensor refers to an individual computation defined by a Function. For
+    // instance, we could create a following tensor given the function above:
+    int output_idx = 0; // Used to index the computation
+    Tensor* X = new Tensor(func, output_idx);
+    std::cout << "Tensor computation: " << *X << std::endl;
+    // Prints: Tensor computation: Tensor X(i[64], j[32]) = i * j
+
+    // Similarly to how we provide a more convenient way of using handles for
+    // constructing Exprs, Tensors also have a more convenient API for
+    // construction. It is based on Compute functions, which take a name:
+    // dimensions, and a lambda specifying the computation body:
+    Tensor* Z = Compute(
+        "Z",
+        {{64, "i"}, {32, "j"}},
+        [](const VarHandle& i, const VarHandle& j) { return i / j; });
+    std::cout << "Tensor computation: " << *Z << std::endl;
+    // Prints: Tensor computation: Tensor Z(i[64], j[32]) = i / j
+
+    // Tensors might access other tensors and external placeholders in their
+    // expressions. It can be done like so:
+    Placeholder P("P", kFloat, {64, 32});
+    Tensor* R = Compute(
+        "R",
+        {{64, "i"}, {32, "j"}},
+        [&](const VarHandle& i, const VarHandle& j) {
+          return Z->call(i, j) * P.load(i, j);
+        });
+    std::cout << "Tensor computation: " << *R << std::endl;
+    // Prints: Tensor computation: Tensor R(i[64], j[32]) = Z(i, j) * P[i, j]
+
+    // Placeholders could be thought of as external tensors, i.e. tensors for
+    // which we don't have the element expression. In other words, for `Tensor`
+    // we know an expression specifying how its elements can be computed (a
+    // mathematical formula). For external tensors, or placeholders, we don't
+    // have such an expression. They need to be considered as coming to us as
+    // inputs from outside - we can only load data from them.
+    //
+    // Also note that we use 'call' to construct an access to an element of a
+    // Tensor and we use 'load' for accessing elements of an external tensor
+    // through its Placeholder. This is an implementation detail and could be
+    // changed in future.
+    //
+    // Why do we have Functions and Tensors and what is the relationship between
+    // them? Functions are used to represent several computations performed over
+    // the same domain. Tensors refer to individual computations of a Function.
```

Comment:
An example illustrating this would be helpful, since that doesn't seem to be what's shown above.